controllers/connectionpool_controller.go

```diff
@@ -219,24 +215,25 @@ func (h ConnectionPoolHandler) checkPreconditions(ctx context.Context, avn *aive
 	meta.SetStatusCondition(&cp.Status.Conditions,
 		getInitializedCondition("Preconditions", "Checking preconditions"))
 
-	check, err := checkServiceIsRunning(ctx, avn, avnGen, cp.Spec.Project, cp.Spec.ServiceName)
+	isRunning, err := checkServiceIsRunning(ctx, avn, avnGen, cp.Spec.Project, cp.Spec.ServiceName)
 	if err != nil {
 		return false, err
 	}
 
-	if check {
-		db, err := avn.Databases.Get(ctx, cp.Spec.Project, cp.Spec.ServiceName, cp.Spec.DatabaseName)
-		if err != nil {
-			if isNotFound(err) {
-				return false, nil
-			}
-			return false, err
-		}
+	if !isRunning {
+		return false, nil
+	}
+
+	_, err = avn.Databases.Get(ctx, cp.Spec.Project, cp.Spec.ServiceName, cp.Spec.DatabaseName)
```

Comment:
There is no single user get method. The GET endpoint returns a list. Using the original client.

Reply:
Is this a potential bug to work on in the codegen client?

Reply:
It's not a bug, there is just no such an endpoint in the API.

Reply:
I think we need to raise this to the relevant stakeholders, because this isn't a first time when this happens. I would recommend making up a list of all of these instances where this is needed.